opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1254,15 +1255,52 @@ namespace detail {
 
     namespace detail
     {
+        /// \brief Compute the L-infinity norm of a vector
+        /// \warn This function is not suitable to compute on the well equations.
+        /// \param a The container to compute the infinity norm on.
+        ///          It has to have one entry for each cell.
+        /// \param info In a parallel this holds the information about the data distribution.
+        double infinityNorm( const ADB& a, const boost::any& pinfo=boost::any() )
+        {
+#if HAVE_MPI
+            if ( pinfo.type() == typeid(ParallelISTLInformation) )
+            {
+                const ParallelISTLInformation& real_info =
+                    boost::any_cast<const ParallelISTLInformation&>(pinfo);
+                double result=0;
+                real_info.computeReduction(a.value(), Reduction::makeGlobalMaxFunctor<double>(), result);
+                return result;
+            }
+            else
+#endif
+            {
+                if( a.value().size() > 0 ) {
+                    return a.value().matrix().lpNorm<Eigen::Infinity> ();
+                }
+                else { // this situation can occur when no wells are present
+                    return 0.0;
+                }
+            }
+        }
 
-        double infinityNorm( const ADB& a )
+        /// \brief Compute the L-infinity norm of a vector representing a well equation.
+        /// \param a The container to compute the infinity norm on.
+        /// \param info In a parallel this holds the information about the data distribution.
+        double infinityNormWell( const ADB& a, const boost::any& pinfo )
         {
+            double result=0;
             if( a.value().size() > 0 ) {
-                return a.value().matrix().lpNorm<Eigen::Infinity> ();
+                result = a.value().matrix().lpNorm<Eigen::Infinity> ();
             }
-            else { // this situation can occur when no wells are present
-                return 0.0;
+#if HAVE_MPI
+            if ( pinfo.type() == typeid(ParallelISTLInformation) )
+            {
+                const ParallelISTLInformation& real_info =
+                    boost::any_cast<const ParallelISTLInformation&>(pinfo);
+                result = real_info.communicator().max(result);
```

Comment:
To see if I understood this:
 - the infinityNormWell() is simpler because it can assume that there are no trash values (values we computed for overlap cells we do not own) to be filtered out during reduction
 - the infinityNorm() cannot be used in the well case because it is designed *only* for the "reduce cell variable(s)" case.

Perhaps we could make this more obvious in documentation. I will not keep this back from merging however.